controllers/control/k8ssandratask_controller.go

```diff
@@ -0,0 +1,212 @@
+		}
+
+		dcTaskKey := client.ObjectKey{
+			Namespace: desiredDcTask.Namespace,
+			Name:      desiredDcTask.Name,
+		}
+		actualDcTask := &cassapi.CassandraTask{}
+		err = remoteClient.Get(ctx, dcTaskKey, actualDcTask)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				logger.Info("Creating DC task", "CassandraTask", dcTaskKey)
+				err = remoteClient.Create(ctx, desiredDcTask)
+				if err != nil {
+					return ctrl.Result{}, err
+				}
+			} else {
+				return ctrl.Result{}, err
+			}
+		} else {
+			if k8Task.Status.Datacenters == nil {
+				k8Task.Status.Datacenters = make(map[string]cassapi.CassandraTaskStatus)
+			}
+			k8Task.Status.Datacenters[dc.Meta.Name] = actualDcTask.Status
+		}
+
+		k8Task.BuildGlobalStatus()
+		err = r.Status().Update(ctx, k8Task)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func filterDcs(kc *k8sapi.K8ssandraCluster, dcNames []string) ([]k8sapi.CassandraDatacenterTemplate, error) {
+
+	if len(dcNames) == 0 {
+		return kc.Spec.Cassandra.Datacenters, nil
+	}
+
+	dcs := make([]k8sapi.CassandraDatacenterTemplate, 0, len(dcNames))
+	for _, dcName := range dcNames {
+		found := false
+		for _, dc := range kc.Spec.Cassandra.Datacenters {
+			if dc.Meta.Name == dcName {
+				dcs = append([]k8sapi.CassandraDatacenterTemplate{dc}, dcs...)
+				found = true
+				break
+			}
+		}
+		if !found {
+			// TODO ignore instead? could be a race maybe
+			return nil, fmt.Errorf("unknown DC %s", dcName)
+		}
+	}
+	return dcs, nil
+}
+
+func newDcTask(k8Task *api.K8ssandraTask, namespace string, dcName string) *cassapi.CassandraTask {
+	dcTask := &cassapi.CassandraTask{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   namespace,
+			Name:        k8Task.Name + "-" + dcName,
+			Annotations: map[string]string{},
+			Labels: map[string]string{
+				k8sapi.NameLabel:                k8sapi.NameLabelValue,
+				k8sapi.PartOfLabel:              k8sapi.PartOfLabelValue,
+				k8sapi.ComponentLabel:           k8sapi.ComponentLabelValueCassandra,
+				k8sapi.CreatedByLabel:           k8sapi.CreatedByLabelValueK8ssandraTaskController,
+				api.K8ssandraTaskNameLabel:      k8Task.Name,
+				api.K8ssandraTaskNamespaceLabel: k8Task.Namespace,
+			},
+		},
+
+		Spec: cassapi.CassandraTaskSpec{
+			Datacenter: v1.ObjectReference{Namespace: namespace, Name: dcName},
+
+			// TODO revisit this once k8ssandra/cass-operator#458 merged
+			ScheduledTime:           k8Task.Spec.Template.ScheduledTime,
+			Jobs:                    k8Task.Spec.Template.Jobs,
+			RestartPolicy:           k8Task.Spec.Template.RestartPolicy,
+			TTLSecondsAfterFinished: k8Task.Spec.Template.TTLSecondsAfterFinished,
+			ConcurrencyPolicy:       k8Task.Spec.Template.ConcurrencyPolicy,
+		},
+	}
+	return dcTask
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *K8ssandraTaskReconciler) SetupWithManager(mgr ctrl.Manager, clusters []cluster.Cluster) error {
+	cb := ctrl.NewControllerManagedBy(mgr).
```

Comment:
You need to filter your own Status updates here. So, ensure generation has updated.